src/Orleans.Core.Abstractions/Cancellation/GrainCancellationToken.cs

```diff
@@ -69,49 +65,43 @@ internal GrainCancellationToken(Guid id, bool canceled) : this(id)
 
         internal Task Cancel()
         {
-            // propagate the exception from the _cancellationTokenSource.Cancel back to the caller
-            // but also cancel _targetGrainReferences. 
-            Task task = OrleansTaskExtentions.WrapInTask(_cancellationTokenSource.Cancel);
-
-            if (_targetGrainReferences.IsEmpty)
+            if (_cancellationTokenRuntime == null)
             {
-                return task;
+                WrapInTask(_cancellationTokenSource.Cancel);
+                return Task.CompletedTask;
```

Comment:
you need to return the result of `WrapInTask`.
BTW, since this is not a general purpose method, just inline the wrapping here and do the try/catch in the same method, no need to create a separate WrapInTask method